### PR TITLE
[fix](mtmv) Fixed an error in setting variables in nested materialize views

### PR DESCRIPTION
**Pull request by felixwluo**
Fixed an error in setting variables in nested materialized views

### What problem does this PR solve?

Issue Number: close #xxx

Recurrence
```
create database test;

use test;

CREATE TABLE tb1 (
        `id` bigint NOT NULL COMMENT '',
        `map_infos` map < int,
        varchar(65533) > NULL COMMENT ''
) ENGINE = OLAP UNIQUE KEY(`id`) COMMENT 'test' DISTRIBUTED BY HASH(`id`) BUCKETS 2 PROPERTIES (
        "replication_allocation" = "tag.location.default: 1",
        "min_load_replica_num" = "-1",
        "is_being_synced" = "false",
        "storage_medium" = "hdd",
        "storage_format" = "V2",
        "inverted_index_storage_format" = "V1",
        "enable_unique_key_merge_on_write" = "true",
        "light_schema_change" = "true",
        "disable_auto_compaction" = "false",
        "enable_single_replica_compaction" = "false",
        "group_commit_interval_ms" = "10000",
        "group_commit_data_bytes" = "134217728",
        "enable_mow_light_delete" = "false"
);

insert into
       tb1
select
        id,
        map_agg(a, b)
from
(
                select
                        123 id,
                        3 a,
                        '5' b
                union
                all
                select
                        123 id,
                        6 a,
                        '8' b
        ) aa
group by
        id;
        
 CREATE MATERIALIZED VIEW mv1 BUILD IMMEDIATE REFRESH COMPLETE ON SCHEDULE EVERY 10 MINUTE DUPLICATE KEY(info_id) DISTRIBUTED BY HASH(`info_id`) BUCKETS 2 PROPERTIES (
        "replication_allocation" = "tag.location.default: 1",
        "min_load_replica_num" = "-1",
        "is_being_synced" = "false",
        "colocate_with" = "dwd_info_group",
        "storage_medium" = "hdd",
        "storage_format" = "V2",
        "inverted_index_storage_format" = "V1",
        "light_schema_change" = "true",
        "disable_auto_compaction" = "false",
        "enable_single_replica_compaction" = "false",
        "group_commit_interval_ms" = "10000",
        "group_commit_data_bytes" = "134217728",
        "enable_nondeterministic_function" = "true"
) AS
select
        /*+ SET_VAR(enable_force_spill = true) */
        cast(a.id as bigint) info_id,
        map_infos
from
        tb1 a;
        
CREATE MATERIALIZED VIEW mv2 BUILD IMMEDIATE REFRESH COMPLETE ON SCHEDULE EVERY 10 MINUTE DUPLICATE KEY(info_id) DISTRIBUTED BY HASH(`info_id`) BUCKETS 2 PROPERTIES (
        "replication_allocation" = "tag.location.default: 1",
        "min_load_replica_num" = "-1",
        "is_being_synced" = "false",
        "colocate_with" = "dwd_info_group",
        "storage_medium" = "hdd",
        "storage_format" = "V2",
        "inverted_index_storage_format" = "V1",
        "light_schema_change" = "true",
        "disable_auto_compaction" = "false",
        "enable_single_replica_compaction" = "false",
        "group_commit_interval_ms" = "10000",
        "group_commit_data_bytes" = "134217728",
        "enable_nondeterministic_function" = "true"
) AS
select
        info_id,
        map_infos
from
        mv1 a;
```

Error
`java.lang.NullPointerException: Cannot invoke "org.apache.doris.nereids.StatementContext.getConnectContext()" because "context" is null`

### CODE REVIEW

**Comment by felixwluo** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/properties/SelectHintSetVar.java [hunk 1]
```diff
@@ -50,6 +50,9 @@ public Map<String, Optional<String>> getParameters() {
      * @param context statement context
      */
     public void setVarOnceInSql(StatementContext context) {
+        if (context == null) {
```
Comment (seawinde) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/properties/SelectHintSetVar.java:53]: I think the root cause maybe is the code in org.apache.doris.mtmv.MTMVCache#from

the code as following
```
        StatementContext mvSqlStatementContext = new StatementContext(connectContext,
                new OriginStatement(mtmv.getQuerySql(), 0));
        NereidsPlanner planner = new NereidsPlanner(mvSqlStatementContext);
        if (mvSqlStatementContext.getConnectContext().getStatementContext() == null) {
            mvSqlStatementContext.getConnectContext().setStatementContext(mvSqlStatementContext);
        }
```

should before` LogicalPlan unboundMvPlan = new NereidsParser().parseSingle(mtmv.getQuerySql())`;

then the StatementContext is not null in current connect Context

Comment (felixwluo) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/properties/SelectHintSetVar.java]: thanks, I have modified it, please help review it.

**Comment by felixwluo** [comment]
run buildall

**Review by seawinde — APPROVED** [review]
LGTM

**Review by morrySnow — APPROVED** [review]

**Review by lide-reed — APPROVED** [review]
LGTM

**Pull request merged**